server/ngwmn/services/ngwmn.py

```diff
@@ -45,6 +43,81 @@ def get_well_lithography(agency_cd, location_id, service_root=SERVICE_ROOT):
     return parse_xml(resp.content)
 
 
+def get_water_quality_activities(agency_cd, location_id):
+    """
+    Retrieves water-quality data from the NGWMN iddata service.
+
+    :param str agency_cd: agency code for the agency that manages the location
+    :param str location_id: the location's identifier
+    :return: array of activity dictionaries
+    :rtype: array
+    """
+    xml = get_iddata('water_quality', agency_cd, location_id)
+    if xml is None:
+        return []
+
+    organization = xml.find('.//Organization', xml.nsmap)
+
+    return [{
```

Comment:
This seems awfully complicated, but I suppose that's the nature of the data we have to work with.

Reply:
Yeah it's pretty dense, although I find using the declarative, inline dict and list comprehensions make it easier to read than for loops with append and assignment operations.

To avoid extending XML-isms into code, I personally like this way of parsing.. it's more explicit than a direct XML-to-dict conversion, and we could be doing other Python conversions here - eg, return date objects rather than strings. But I'm certainly open to using other approaches.